Sprint-1/implement/sum.test.js

```diff
@@ -13,24 +13,40 @@ const sum = require("./sum.js");
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+describe("sum", () => {
+    test("returns 0 with empty array given", () => {
+        expect(sum([])).toEqual(0);
+    });
+
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
-
+test("returns a number with just one number array given", () => {
+    expect(sum([2])).toEqual(2);
+});
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
-
+test("returns  total sum with an array containing negative numbers given", () => {
+    expect(sum([-2, -6, -8])).toEqual(-16);
+});
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
-
+test("returns  total sum with an array decimal/float numbers given", () => {
+    expect(sum([1.2, 0.6, 0.005])).toEqual(1.8049999999999997);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`. Even changing the order in which the program add/subtract numbers can yield different values.

So the following could happen
```
  expect(sum([1.2, 0.6, 0.005])).toEqual(1.8049999999999997);   // This pass
  expect(sum([0.005, 0.6, 1.2])).toEqual(1.8049999999999997);   // This fail
```

**Can you find a more appropriate way to test a value (that involves decimal number calculations) for equality?**